training/controllers/all_controllers.py

```diff
@@ -84,5 +114,30 @@ def logout():
 
 
 @app.route('/ver')
+@login_required
 def ver():
-    return sesion.get('username')
\ No newline at end of file
+    return sesion.get('username')
+
+
+@app.route('/mail')
+def prueba_mail():
+    email = 'donnetta7@adriveriep.com'
+    msg = Message('Confirm Email', sender='anthony@prettyprinted.com', recipients=[email])
+    msg.body = "Bienvenido"
+    mail.send(msg)
+    return "0"
+
+@app.route('/mail/validation/<string:username>')
+def email_verification(username):
+    our_user = session.query(User).filter_by(id=username).first()
+
+    if our_user is None:
+        return "No existe ese usuario"
+    else:
+        if our_user.mail_validation:
+            return "El email de {} ya estaba validado".format(our_user.username)
+        else:
+            #No me gusta esta manera de updatear
+            our_user.mail_validation = True
+            session.commit()
+            return "Se valido el email de {}".format(our_user.username)
```

Comment:
You should use a random token saved to the database or a encrypted token string as validation parameter. Using the username is easy to bypass by anyone.

Reply:
It was released a new feature called "recover". Used to recover the mail. That feature was developed with encrypted tokens and it is in charge to send emails to people who forgot their passwords.